app/scripts/lib/PatchStore.ts

```diff
@@ -79,25 +85,46 @@ export class PatchStore {
     }
   }
 
-  private _generatePatches(
-    oldState: Record<string, unknown>,
-    newState: Record<string, unknown>,
-  ): Patch[] {
-    return Object.keys(newState)
-      .map((key) => {
-        const oldData = oldState[key];
-        const newData = newState[key];
-
-        if (oldData === newData) {
-          return null;
-        }
-
-        return {
-          op: 'replace',
-          path: [key],
-          value: newData,
-        };
-      })
-      .filter(Boolean) as Patch[];
+  private _generatePatches({
+    controllerKey,
+    oldState,
+    newState,
+  }: {
+    controllerKey?: keyof BackgroundStateProxy;
+    oldState: BackgroundStateProxy;
+    newState: BackgroundStateProxy;
+  }): Patch[] {
+    return controllerKey
+      ? getKnownPropertyNames(newState[controllerKey]).reduce<Patch[]>(
+          (patches, key) => {
+            // @ts-expect-error There's no way to narrow `key` to a type that corresponds to the current iteration's `controllerKey` value.
+            const oldData = oldState[controllerKey][key];
```

Comment:
Is the old and new state nested within the controller key?

Looking at the `ComposableObservableStore`, it seems the event properties still contain a single controller state object?